internal/kinds/capp-revision/controllers/revision_controller.go

```diff
@@ -0,0 +1,105 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	cappv1alpha1 "github.com/dana-team/container-app-operator/api/v1alpha1"
+	"github.com/dana-team/container-app-operator/internal/kinds/capp-revision/action_mangers"
+	"github.com/dana-team/container-app-operator/internal/kinds/capp-revision/adapters"
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	"reflect"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"time"
+)
+
+const RequeueTime = 5 * time.Second
+
+// CappRevisionReconciler reconciles a Capp object
+type CappRevisionReconciler struct {
+	Log logr.Logger
+	client.Client
+	Scheme        *runtime.Scheme
+	EventRecorder record.EventRecorder
+}
+
+func (r *CappRevisionReconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	logger := log.FromContext(ctx).WithValues("CappName", req.Name, "CappNamespace", req.Namespace)
+	logger.Info("Starting Reconcile")
+	capp := cappv1alpha1.Capp{}
+	if err := r.Client.Get(ctx, req.NamespacedName, &capp); err != nil {
+		if errors.IsNotFound(err) {
+			logger.Info(fmt.Sprintf("Didn't find Capp: %s, from the namespace: %s", capp.Name, capp.Namespace))
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, fmt.Errorf("failed to get Capp: %s", err.Error())
+	}
+	if !capp.DeletionTimestamp.IsZero() {
+		return ctrl.Result{}, nil
+	}
+	if err := syncCappRevision(ctx, r.Client, capp, logger); err != nil {
+		if errors.IsConflict(err) {
+			logger.Info(fmt.Sprintf("Conflict detected requeuing: %s", err.Error()))
+			return ctrl.Result{RequeueAfter: RequeueTime}, nil
+		}
+		return ctrl.Result{}, fmt.Errorf("failed to sync Capp: %s", err.Error())
+	}
+	return ctrl.Result{}, nil
+}
+
+// syncCappRevision manages the lifecycle of CappRevisions based on the state of a Capp, handling creation, update, or deletion.
+func syncCappRevision(ctx context.Context, k8sClient client.Client, capp cappv1alpha1.Capp, logger logr.Logger) error {
+	cappRevisions, err := adapters.GetCappRevisions(ctx, k8sClient, capp)
+	if err != nil {
+		logger.Error(err, "could not fetch cappRevisions")
+		return err
+	}
+
+	if len(cappRevisions) == 0 && capp.DeletionTimestamp.IsZero() {
```

Comment:
We don't actually need to check here the DeletionTimestamp of Capp since it it's non-zero we wouldn't arrive at this function in the first place